### PR TITLE
add tests for algob sign-multisig

### PR DESCRIPTION
**Pull request by ratik21**
## Proposed Changes

Added tests for `algob sign-multisig` command. Removed a TODO from previous test.

### CODE REVIEW

**Comment by amityadav0** [comment]
do we have test for `signMultiSig` function ?
this doesn't seems to test that.

**Comment by amityadav0** [comment]
with new types i cannot read `msig` property here.
could you please try it with new types ?

**Comment by ratik21** [comment]
I think we discussed that `decodeSignedTransaction` should return `msig` property. In tests with latest `algosdk` (although without types) i am getting the `txn.msig` property (as expected). So it seems like a types issue right (same as [https://github.com/algorand/js-algorand-sdk/issues/389](https://github.com/algorand/js-algorand-sdk/issues/389))?

**Comment by amityadav0** [comment]
> I think we discussed that `decodeSignedTransaction` should return `msig` property. In tests with latest `algosdk` (although without types) i am getting the `txn.msig` property (as expected). So it seems like a types issue right (same as [algorand/js-algorand-sdk#389](https://github.com/algorand/js-algorand-sdk/issues/389))? Msig should be there in new types

I know we discussed, but need to confirm does this work here.

File: packages/algob/test/builtin-tasks/sign-multisig.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  });
+
+  it("Append bob's signature to multisigned txn loaded from file", async function () {
+    const encodedTx = loadSignedTxnFromFile(inputFile) as Uint8Array;
+    const txMsig = decodeSignedTransaction(encodedTx).msig;
+
+    // input assertions (verfiy bob pk is present in pre-image, but it is not signed,
+    // i.e signature is not present)
+    assert.isDefined(txMsig);
+    assert.deepInclude(txMsig?.subsig, { pk: bobPk } as any); // msig includes bobPk (but without signature i.e "s" field)
```
Comment (amityadav0) [inline packages/algob/test/builtin-tasks/sign-multisig.ts:45]: do we need this part ? can we remove it ? or  move to another section.

File: packages/algob/test/builtin-tasks/sign-multisig.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      file: inputFile,
+      account: 'bob',
+      out: outFile
+    });
+
+    assert.isTrue(stub.calledWith(
+     `File assets/${outFile} already exists. Aborting. Use --force flag if you want to overwrite it`));
+  });
+
+  it("Should append _out to input file if outfile is not passed", async function () {
```
Comment (amityadav0) [inline packages/algob/test/builtin-tasks/sign-multisig.ts:118]:
```suggestion
  it("Should append output to input file if outfile is not passed", async function () {
```

File: packages/algob/test/builtin-tasks/sign-multisig.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    await this.env.run(TASK_SIGN_MULTISIG, {
+      file: inputFile,
+      account: 'bob'
+    });
+
+    const newOutPath = path.join(ASSETS_DIR, 'multisig-signed_out.txn');
+    assert.isTrue(fs.existsSync(newOutPath)); // outfile path (with appended _out) should exist
+
```
Comment (amityadav0) [inline packages/algob/test/builtin-tasks/sign-multisig.ts:126]: how about adding assertions here that output file has multisigned bob sig ?

File: packages/algob/test/builtin-tasks/sign-multisig.ts (hunk 1 above)
Comment (ratik21) [reply packages/algob/test/builtin-tasks/sign-multisig.ts:45]: yes, we need it. We're trying to append bob signature to multisig, so we ensure first it's not present when loading the multisig from file. i.e `pk` is present but `s` isn't.

File: packages/algob/test/builtin-tasks/sign-multisig.ts (hunk 2 above)
Comment (ratik21) [reply packages/algob/test/builtin-tasks/sign-multisig.ts]: NO. this test is that if output file name is not passed, then input filename is appended `_out`. Reverting  the commit

File: packages/algob/test/builtin-tasks/sign-multisig.ts (hunk 3 above)
Comment (ratik21) [reply packages/algob/test/builtin-tasks/sign-multisig.ts:126]: we're already asserting this in first test - [https://github.com/scale-it/algo-builder/pull/410/files#diff-53b33f6a4c7dc9293cd4572a9584c30ac6e217826adf4406e6a7c9eb2a05eca1R38](https://github.com/scale-it/algo-builder/pull/410/files#diff-53b33f6a4c7dc9293cd4572a9584c30ac6e217826adf4406e6a7c9eb2a05eca1R38). 

This test is to verify input file name is append `_out` str.

File: packages/algob/test/builtin-tasks/sign-multisig.ts (hunk 2 above)
Comment (amityadav0) [reply packages/algob/test/builtin-tasks/sign-multisig.ts:118]: could you please update the info, that -out is appended to file name ?
It seems from line that output is appended

Comment (ratik21) [reply packages/algob/test/builtin-tasks/sign-multisig.ts:118]: ok

**Review by amityadav0 — APPROVED** [review]

**Pull request merged**